### PR TITLE
Package and base image update

### PR DESCRIPTION
**Pull request by finestructure**
We're now building and running on Swift 5.4. Let's run this on staging today and deploy onwards tomorrow.

No idea how the test previously even compiled - but I'm glad at least the tests failed. I'd like to understand how this happened but I'm at a loss at the moment how to debug this.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -29,8 +29,8 @@ docker-push:
 
 test-docker:
 	@# run tests inside a docker container
-	docker run --rm -v "$(PWD)":/host -w /host --network="host" finestructure/spi-base:0.3.1 \
-	  bash -c "apt-get update && apt-get install -y unzip && make test"
```
Comment (finestructure) [inline Makefile:33]: Could you give `make test-docker` on your machine another spin please to confirm it's all working locally as well?

**Comment by finestructure** [comment]
Oh wow, TIL: `Package.resolved` is not the definitive build source. What _actually_ happens if you have this in `Package.resolved`:

```
        "package": "Plot",
        "repositoryURL": "https://github.com/JohnSundell/Plot.git",
        "state": {
          "branch": null,
          "revision": "c0bdc94c552bba1b0887103ebba6e20bfdf1f8d9",
          "version": "0.9.3"
        }
```

and compile with Swift 5.3 is that it resolves back to 0.8:

```
❯ env DEVELOPER_DIR="/Applications/Xcode_12.4.app" xcrun swift build
Fetching https://github.com/apple/swift-package-manager.git
...
Cloning https://github.com/JohnSundell/Plot.git
Resolving https://github.com/JohnSundell/Plot.git at 0.8.0
...
❯ cat Package.swift
```

```
...
        "package": "Plot",
        "repositoryURL": "https://github.com/JohnSundell/Plot.git",
        "state": {
          "branch": null,
          "revision": "61e828949ca6f84071bde65c8fc046cf74b7d1a2",
          "version": "0.8.0"
        }
...
```

And that's why the test then failed - back to the 0.8.0 behaviour.

**Comment by finestructure** [comment]
We should probably fail CI if `Package.resolved` has changes after the build step.

File: Makefile (hunk 1 above)
Comment (daveverwer) [reply Makefile:33]: I get a rather unusual error:

```
> make test-docker
docker run --rm -v "/Users/dave/Developer/SwiftPackageIndex/Server":/host -w /host --network="host" finestructure/spi-base:0.4.0 \
	  make test
swift test --enable-test-discovery --enable-code-coverage
warning: '--enable-test-discovery' option is deprecated; tests are automatically discovered on all platforms
[1/1] Planning build
[1/1] Planning build

* Build Completed!
[1/1076] Compiling Logging LogHandler.swift
[2/1076] Compiling Logging Locks.swift
[3/1076] Compiling Logging Logging.swift
[4/1076] Compiling Plot ComponentBuilder.swift
[5/1076] Compiling Plot ComponentContainer.swift
[6/1076] Compiling Plot HTMLComponents.swift
[7/1076] Compiling Plot HTMLElements.swift
[8/1076] Compiling Plot HTMLFormContentType.swift
[9/1076] Compiling Plot HTMLFormMethod.swift
ERRO[0022] error waiting for container: invalid character 'u' looking for beginning of value 
make: *** [test-docker] Error 125
```

Comment (daveverwer) [reply Makefile:33]: Although, something may be up 😬

I run it again and get this:

```
> make test-docker
docker run --rm -v "/Users/dave/Developer/SwiftPackageIndex/Server":/host -w /host --network="host" finestructure/spi-base:0.4.0 \
	  make test
docker: Error response from daemon: Bad response from Docker engine.
See 'docker run --help'.
make: *** [test-docker] Error 125
```

Also, it seems to have deleted all my Docker images 🚀

Comment (daveverwer) [reply Makefile:33]: Even after a reboot:

```
> make test-docker
docker run --rm -v "/Users/dave/Developer/SwiftPackageIndex/Server":/host -w /host --network="host" finestructure/spi-base:0.4.0 \
	  make test
docker: Error response from daemon: dial unix docker.raw.sock: connect: connection refused.
See 'docker run --help'.
make: *** [test-docker] Error 125
```

Something is horribly wrong with my Docker. I'll reinstall it.

**Review by daveverwer — APPROVED** [review]
All tests passing locally and in Docker.

**Pull request merged**